src/types/timestamp.rs

```diff
@@ -75,10 +75,10 @@ fn number_of_digits(value: u32) -> u32 {
 /// of an arbitrary unit.
 #[derive(Debug, Clone)]
 pub struct Timestamp {
-    date_time: NaiveDateTime,
-    offset: Option<FixedOffset>,
-    precision: Precision,
-    fractional_seconds: Option<Mantissa>,
+    pub(crate) date_time: NaiveDateTime,
+    pub(crate) offset: Option<FixedOffset>,
+    pub(crate) precision: Precision,
+    pub(crate) fractional_seconds: Option<Mantissa>,
```

Comment:
I think these are only being made visible because you added an inherent impl to a different module. Is that right?

Reply:
Yeah. Even if it's not an inherent impl (e.g. an ext trait) we'd need access. Open to other ideas..

Reply:
This suggests that implementing `TimeLike` and `DateLike` (as well as some hypothetical `IonTimestampLike` or some such) are probably in order for the API.

Temporarily to work around the fact that we do not expose a useful API, this seems okay.

Reply:
I can open an issue to track that?

Reply:
Please :D

Reply:
Did you one better and added #273.